### PR TITLE
Closing JarInputStream on failure. fixes #1184

### LINKED ISSUE #1184: Potential resource leak in SelfContainedJarFile
SelfContainedJarFile doesn't close `JarInputStream`.
https://github.com/embulk/embulk/blob/master/embulk-core/src/main/java/org/embulk/deps/SelfContainedJarFile.java#L60

This is a potential resource leak.

### PR DESCRIPTION
**Pull request by shroman**
Closing JarInputStream on failure.

### CODE REVIEW

File: embulk-core/src/main/java/org/embulk/deps/SelfContainedJarFile.java [hunk 1]
```diff
@@ -55,7 +55,7 @@
             return;
         }
 
-        final JarInputStream jarInputStream;
+        JarInputStream jarInputStream = null;
```
Comment (dmikurube) [inline embulk-core/src/main/java/org/embulk/deps/SelfContainedJarFile.java]: Did you have special reasons why we needed to remove `final` and assign `null` here?

File: embulk-core/src/main/java/org/embulk/deps/SelfContainedJarFile.java [hunk 2]
```diff
@@ -64,6 +64,7 @@
             this.manifest = null;
             this.innerResources = null;
             this.innerResourcesBinary = null;
+            closeQuiet(jarInputStream);
```
Comment (dmikurube) [inline embulk-core/src/main/java/org/embulk/deps/SelfContainedJarFile.java]: If the execution goes into this `catch` block, it means that it failed in constructing an instance of `JarInputStream`, then `jarInputStream` is not assigned. I don't think we need to close it, then?

Comment (shroman) [reply embulk-core/src/main/java/org/embulk/deps/SelfContainedJarFile.java]: you are right, I missed the place. Modified.

File: embulk-core/src/main/java/org/embulk/deps/SelfContainedJarFile.java (hunk 1 above)
Comment (shroman) [reply embulk-core/src/main/java/org/embulk/deps/SelfContainedJarFile.java]: Not needed if the next issue you raised is addressed. thanks.

**Pull request merged**